lib/macho/macho_file.rb

```diff
@@ -170,6 +170,33 @@ def command(name)
 
     alias :[] :command
 
+    # Add a new load command to the Mach-O.
+    # @param lc [MachO::LoadCommand] the load command being added
+    # @return [void]
+    # @raise [MachO::LoadCommandNotSerializableError] if the command can't be serialized
+    # @raise [MachO::HeaderPadError] if the new command exceeds the header pad buffer
+    # @note This is public, but methods like {#add_rpath} should be preferred.
+    def add_command(lc)
+      raise LoadCommandNotSerializableError.new(LOAD_COMMANDS[lc]) unless lc.serializable?
```

Comment:
Is this check actually needed? I think it is redundant as the `LoadCommand#serialize` invocation below will perform the exact same check.